fragments/infra-boot.sh

```diff
@@ -1,86 +1,253 @@
+
+# Select the EPEL release to make it easier to update
+EPEL_RELEASE_VERSION=7-5
+
+# Send success status to OpenStack WaitCondition
 function notify_success() {
-    $WC_NOTIFY --data-binary  "{\"status\": \"SUCCESS\", \"reason\": \"$1\", \"data\": \"$1\"}"
+    $WC_NOTIFY --data-binary \
+               "{\"status\": \"SUCCESS\", \"reason\": \"$1\", \"data\": \"$1\"}"
     exit 0
 }
 
+# Send success status to OpenStack WaitCondition
 function notify_failure() {
-    $WC_NOTIFY --data-binary "{\"status\": \"FAILURE\", \"reason\": \"$1\", \"data\": \"$1\"}"
+    $WC_NOTIFY --data-binary \
+               "{\"status\": \"FAILURE\", \"reason\": \"$1\", \"data\": \"$1\"}"
     exit 1
 }
 
-#[ -e /run/ostree-booted ] && notify_success "OpenShift node has been prepared for running ansible."
-#systemctl is-enabled os-collect-config || notify_failure "os-collect-config service is not installed or enabled"
+# 
+# --- DNS functions ----------------------------------------------------------
+#
+# true if local DNS service is enabled
+function use_local_dns() {
+    [ "$SKIP_DNS" != "true" ]
+}
+
+# add the local nameserver to the beginning of the local resolver list
+function add_nameserver() {
+    # NAMESERVER_IP=$1
+    sed -i '/search openstacklocal.*/a\nnameserver $1/' /etc/resolv.conf
+}
+
+# Disable automatic updates of resolv.conf when an interface comes up
+function disable_resolv_updates() {
+    # INTERFACE=$1
+    sed -i -e '/^PEERDNS=/s/=.*/="no"/' \
+        /etc/sysconfig/network-scripts/ifcfg-$1
+}
 
-# master and nodes
-sed -i -e 's/^PEERDNS.*/PEERDNS="no"/' /etc/sysconfig/network-scripts/ifcfg-eth0
 
-if [ -e /run/ostree-booted ]; then
-    # Set the DNS to the one provided
-    cp /etc/resolv.conf /etc/resolv.conf.local
-    # FIXME - the clean way would be to set nameserver for openstack's neutron network
-    [ "$SKIP_DNS" = "true" ] || sed -i 's/search openstacklocal.*/&\nnameserver $DNS_IP/' /etc/resolv.conf
+# ----------------------------------------------------------------------------
+# Functions for Atomic Host systems
+# ----------------------------------------------------------------------------
 
-    HOME=/root
-    mv /etc/systemd/system/multi-user.target.wants/docker-storage-setup.service $HOME
+# check if this is an Atomic host
+function is_atomic_host() {
+    [ -e /run/ostree-booted ]
+}
+
+# remove the docker storage setup service link and re-load the systemd config
+function systemd_docker_disable_storage_setup() {
+    mv /etc/systemd/system/multi-user.target.wants/docker-storage-setup.service /root
     systemctl daemon-reload
+}
 
-    # Setup Docker Storage Volume Group
-    if ! [ -b /dev/vdb ]; then
-      echo "ERROR: device /dev/vdb does not exist" >&2
-      notify_failure "device /dev/vdb does not exist"
+# All hosts must have an external disk device (cinder?) for docker storage
+function docker_check_for_storage_device() {
+    # DOCKER_VOLUME_DEVICE=$1
+    if [ -b $1 ]
```

Comment:
there is a missing "!" before test which makes this test failing if vdb is present